[Bug] Project replan REST endpoint skips the TeamHarness _safe_id validation for taskId || 项目 replan REST 接口缺少 taskId 安全校验

### Bug Description

What happened:
`POST /api/v1/projects/{id}/replan` accepts **any** `taskId` string and persists it into the project DAG. The handler's own comment claims `_normalize_task` parity with the TeamHarness reference implementation (`plugins/teamharness/mcp/server.py`), but the reference validates `taskId` through `_safe_id` (`[A-Za-z0-9][A-Za-z0-9._-]*`) while the Go port (`normalizeReplanTasks` in `internal/server/project_handler.go`) only checks for an empty string.

What I expected:
Unsafe task ids (spaces, `/`, `..`, non-alphanumeric leading char, Unicode, ...) are rejected with `400`, exactly like:
1. the TeamHarness reference `_normalize_task` → `_safe_id` raises `ValueError("taskId must be a safe id")`, and
2. the sibling endpoint `CreateProject` in the same file, which already validates `project_id` via `isPlainToken`.

Why it matters:
- Task ids flow unsanitized into TaskMeta object keys: `taskMetaKeys()` builds `shared/tasks/{taskID}/meta.json` (and `teams/{team}/shared/tasks/{taskID}/meta.json`). Ids like `t/1` or `../escape` produce traversal-shaped keys.
- `GetTaskArtifact`'s path allowlist uses `strings.HasPrefix(path, "shared/tasks/"+taskID+"/")`; a crafted `taskID` weakens that check.
- Every TeamHarness MCP operation on such tasks (`_safe_id` is applied again downstream) fails afterwards, leaving permanently unmanageable zombie tasks in the DAG.

### Steps to Reproduce

Against any controller with the projects API (`feat(projects)` #1172, commit `8dd02667` or later):

```bash
curl -X POST "$CONTROLLER/api/v1/projects/p1/replan" \
  -H "Authorization: Bearer $TOKEN" \
  -d '{"tasks":[{"taskId":"../escape","title":"X"}]}'
# => 200 OK, task persisted. Reference implementation rejects it.
```

Unit-level reproduction (fails before the fix):

```go
body := `{"tasks":[{"taskId":"../escape","title":"X"}]}`
// ReplanProject returns 200; TeamHarness _safe_id would reject this id.
```

Accepted-but-invalid ids verified on `332d57a4`: `"task with space"`, `"../escape"`, `"t/1"`, `"-leading-dash"`, `"任务一"` — all return `200`.

### AI Analysis

Found by code audit of the new projects write API (#1172): comparing `normalizeReplanTasks` (Go) against `_normalize_task`/`_safe_id` (Python reference) shows the safe-id validation was dropped in the port. No runtime logs needed — the divergence is deterministic and covered by the regression test in the linked PR.

### Relevant Logs

```shell
=== RUN   TestReplanProject_UnsafeTaskID400
    taskId "task with space": status=200, want 400 (reference _safe_id rejects it)
    taskId "../escape": status=200, want 400 (reference _safe_id rejects it)
    taskId "t/1": status=200, want 400 (reference _safe_id rejects it)
    taskId "-leading-dash": status=200, want 400 (reference _safe_id rejects it)
    taskId "任务一": status=200, want 400 (reference _safe_id rejects it)
--- FAIL: TestReplanProject_UnsafeTaskID400
```

### Component

Other

### Version / Commit

332d57a4

<!--This is a translation content dividing line, the content below is generated by machine, please do not modify the content below-->
---
### Bug 描述

现象：`POST /api/v1/projects/{id}/replan` 接受任意 `taskId` 字符串并写入项目 DAG。该处理函数的注释声称与 TeamHarness 参考实现（`plugins/teamharness/mcp/server.py`）的 `_normalize_task` 保持一致，但参考实现通过 `_safe_id`（`[A-Za-z0-9][A-Za-z0-9._-]*`）校验 `taskId`，而 Go 移植版（`internal/server/project_handler.go` 的 `normalizeReplanTasks`）只检查了空字符串。

期望：不安全的 task id（空格、`/`、`..`、非字母数字开头、Unicode 等）应像以下两处一样返回 `400` 被拒绝：
1. TeamHarness 参考实现 `_normalize_task` → `_safe_id` 会抛出 `ValueError("taskId must be a safe id")`；
2. 同文件的姊妹接口 `CreateProject` 已经用 `isPlainToken` 校验 `project_id`。

影响：
- task id 未经清洗直接进入 TaskMeta 对象 key：`taskMetaKeys()` 拼出 `shared/tasks/{taskID}/meta.json`。形如 `t/1`、`../escape` 的 id 会产生带路径穿越形态的 key；
- `GetTaskArtifact` 的路径白名单使用 `strings.HasPrefix(path, "shared/tasks/"+taskID+"/")`，畸形 taskID 会削弱该校验；
- TeamHarness 侧所有后续 MCP 操作都会再次触发 `_safe_id` 而失败，DAG 中会残留永远无法管理的僵尸任务。

### 复现步骤

见上方 curl / 单元测试。已在 `332d57a4` 验证：`"task with space"`、`"../escape"`、`"t/1"`、`"-leading-dash"`、`"任务一"` 均返回 200。

### 组件

Other

### 版本 / Commit

332d57a4

--- a/agentteams-controller/internal/server/project_handler.go
+++ b/agentteams-controller/internal/server/project_handler.go
@@ -2123,6 +2123,25 @@ func (h *ProjectHandler) ReplanProject(w http.ResponseWriter, r *http.Request) {
 	httputil.WriteJSON(w, http.StatusOK, h.buildWorkflow(meta, pwc.team, false))
 }
 
+// isSafeTaskID mirrors TeamHarness _safe_id ([A-Za-z0-9][A-Za-z0-9._-]*).
+// Task ids become segments of TaskMeta object keys
+// (shared/tasks/{id}/meta.json) and are embedded in the GetTaskArtifact
+// path allowlist, so separators and traversal characters must be rejected.
+func isSafeTaskID(s string) bool {
+	if s == "" {
+		return false
+	}
+	for i, r := range s {
+		switch {
+		case r >= 'a' && r <= 'z', r >= 'A' && r <= 'Z', r >= '0' && r <= '9':
+		case i > 0 && (r == '.' || r == '_' || r == '-'):
+		default:
+			return false
+		}
+	}
+	return true
+}
+
 // normalizeReplanTasks converts raw plan tasks into projectTaskMeta with the
 // same field normalization as TeamHarness _normalize_task: it accepts
 // taskId/task_id, assignedTo/assigned_to, dependsOn/depends_on; status
@@ -2136,9 +2155,12 @@ func normalizeReplanTasks(raw []json.RawMessage, previous map[string]projectTask
 		if err := json.Unmarshal(item, &m); err != nil {
 			return nil, fmt.Errorf("invalid task entry: %v", err)
 		}
-		taskID := firstString(m["taskId"], m["task_id"])
+		taskID := strings.TrimSpace(firstString(m["taskId"], m["task_id"]))
 		if taskID == "" {
 			return nil, fmt.Errorf("taskId is required")
+		}
+		if !isSafeTaskID(taskID) {
+			return nil, fmt.Errorf("taskId must be a safe id: %s", taskID)
 		}
 		prev, hasPrev := previous[taskID]
 		status := firstString(m["status"])

--- a/agentteams-controller/internal/server/project_handler_test.go
+++ b/agentteams-controller/internal/server/project_handler_test.go
@@ -3245,6 +3245,56 @@ func TestReplanProject_DuplicateID400(t *testing.T) {
 	}
 }
 
+// TestReplanProject_UnsafeTaskID400 guards _safe_id parity: the TeamHarness
+// reference rejects task ids outside [A-Za-z0-9][A-Za-z0-9._-]* because ids
+// become TaskMeta object-key segments (shared/tasks/{id}/meta.json).
+func TestReplanProject_UnsafeTaskID400(t *testing.T) {
+	cases := []string{
+		"task with space",
+		"../escape",
+		"t/1",
+		"-leading-dash",
+		".leading-dot",
+		"任务一",
+	}
+	for _, tc := range cases {
+		store := ossfake.NewMemory()
+		putProject(store, "shared/projects/p1/meta.json", map[string]any{
+			"project_id": "p1", "title": "P1", "status": "active", "plan_type": "dag", "tasks": []map[string]any{},
+		})
+		h := newProjectTestHandler(t, store)
+		body := `{"tasks":[{"taskId":"` + tc + `","title":"X"}]}`
+		req := httptest.NewRequest(http.MethodPost, "/api/v1/projects/p1/replan", strings.NewReader(body))
+		req.SetPathValue("id", "p1")
+		req = withCaller(req, &authpkg.CallerIdentity{Role: authpkg.RoleAdmin, Username: "admin"})
+		rec := httptest.NewRecorder()
+		h.ReplanProject(rec, req)
+		if rec.Code != http.StatusBadRequest {
+			t.Errorf("taskId %q: status=%d, want 400 (unsafe id)", tc, rec.Code)
+		}
+	}
+}
+
+// TestReplanProject_SafeTaskIDAccepted guards the positive side of the
+// _safe_id rule: ids mixing letters, digits, dots, dashes and underscores
+// (with a leading alphanumeric) remain valid.
+func TestReplanProject_SafeTaskIDAccepted(t *testing.T) {
+	store := ossfake.NewMemory()
+	putProject(store, "shared/projects/p1/meta.json", map[string]any{
+		"project_id": "p1", "title": "P1", "status": "active", "plan_type": "dag", "tasks": []map[string]any{},
+	})
+	h := newProjectTestHandler(t, store)
+	body := `{"tasks":[{"taskId":"t-1.x_2","title":"X"}]}`
+	req := httptest.NewRequest(http.MethodPost, "/api/v1/projects/p1/replan", strings.NewReader(body))
+	req.SetPathValue("id", "p1")
+	req = withCaller(req, &authpkg.CallerIdentity{Role: authpkg.RoleAdmin, Username: "admin"})
+	rec := httptest.NewRecorder()
+	h.ReplanProject(rec, req)
+	if rec.Code != http.StatusOK {
+		t.Fatalf("status=%d body=%s, want 200 for safe id", rec.Code, rec.Body.String())
+	}
+}
+
 func TestReplanProject_UnknownDependency400(t *testing.T) {
 	store := ossfake.NewMemory()
 	putProject(store, "shared/projects/p1/meta.json", map[string]any{

